experiment/commenter/main.go

```diff
@@ -0,0 +1,167 @@
+	"log"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func flagOptions() options {
+	o := options{}
+	flag.StringVar(&o.query, "query", "", "See https://help.github.com/articles/searching-issues-and-pull-requests/")
+	flag.DurationVar(&o.updated, "updated", 2*time.Hour, "Filter to issues unmodified for at least this long if set")
+	flag.BoolVar(&o.includeClosed, "include-closed", false, "Match closed issues if set")
+	flag.BoolVar(&o.wet, "wet", false, "Mutate github if set")
+	flag.StringVar(&o.comment, "comment", "", "Append the following comment to matching issues")
+	flag.IntVar(&o.ceiling, "ceiling", 3, "Maximum number of issues to modify, 0 for infinite")
+	flag.StringVar(&o.user, "user", "", "Github username of commenter")
+	flag.StringVar(&o.token, "token", "", "Path to github token")
+	flag.Parse()
+	return o
+}
+
+func parseHTMLURL(url string) (string, string, int, error) {
+	// https://github.com/batterseapower/pinyin-toolkit/issues/132`)
+	re := regexp.MustCompile(`.+/(.+)/(.+)/(issues|pull)/(\d+)$`)
+	mat := re.FindStringSubmatch(url)
+	if mat == nil {
+		return "", "", 0, fmt.Errorf("failed to parse: %s", url)
+	}
+	n, err := strconv.Atoi(mat[4])
+	if err != nil {
+		return "", "", 0, err
+	}
+	return mat[1], mat[2], n, nil
+}
+
+func makeQuery(query string, includeClosed bool, minUpdated time.Duration) string {
+	parts := []string{query}
+	if !includeClosed {
+		parts = append(parts, "is:open")
+	}
+	if minUpdated != 0 {
+		latest := time.Now().Add(-minUpdated)
+		parts = append(parts, "updated:<="+latest.Format(time.RFC3339))
+	}
+	return strings.Join(parts, " ")
+}
+
+type options struct {
+	asc           bool
+	ceiling       int
+	comment       string
+	includeClosed bool
+	query         string
+	sort          string
+	token         string
+	updated       time.Duration
+	user          string
+	wet           bool
+}
+
+type client interface {
+	CreateComment(owner, repo string, number int, comment string) error
+	FindIssues(query, sort string, asc bool) ([]github.Issue, error)
+}
+
+func main() {
+	log.SetFlags(log.LstdFlags | log.Lshortfile)
+	o := flagOptions()
+
+	if o.query == "" {
+		log.Fatal("empty --query")
+	}
+	if o.user == "" {
+		log.Fatal("empty --user")
+	}
+	if o.token == "" {
+		log.Fatal("empty --token")
+	}
+	b, err := ioutil.ReadFile(o.token)
+	if err != nil {
+		log.Fatalf("cannot read --token: %v", err)
+	}
+
+	var c client
+	tok := strings.TrimSpace(string(b))
+	if o.wet {
+		c = github.NewClient(o.user, tok)
+	} else {
+		c = github.NewDryRunClient(o.user, tok)
+	}
+
+	query := makeQuery(o.query, o.includeClosed, o.updated)
+	sort := ""
+	asc := false
+	if o.updated > 0 {
+		sort = "updated"
+		asc = true
+	}
+	if err := run(c, query, sort, asc, o.comment, o.ceiling); err != nil {
+		log.Fatalf("Failed run: %v", err)
+	}
+}
+
+func run(c client, query, sort string, asc bool, comment string, ceiling int) error {
+	log.Printf("Searching: %s", query)
+	issues, err := c.FindIssues(query, sort, asc)
```

Comment:
There's an important [TODO](https://github.com/kubernetes/test-infra/blob/b59e249f7821090ade1d0949dc09a5d18fe17423/prow/github/client.go#L746) for this method that should really ought to be fixed before more people use it.